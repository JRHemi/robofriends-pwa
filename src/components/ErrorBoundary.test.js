import React from "react";
import { shallow } from "enzyme";
import ErrorBoundary from "./ErrorBoundary";
// import { describe } from "node:test";

describe("ErrorBoundary", () => {

  it("expects to render Child component", () => {
    const MockChild = () => {
      return null;
    };

    const wrapper = shallow(<ErrorBoundary><MockChild /></ErrorBoundary>);
    expect(wrapper).toMatchSnapshot();
  });
  
  it("logs error", () => {
    const MockChild = () => {
      return null;
    };

    const wrapper = shallow(<ErrorBoundary><MockChild /></ErrorBoundary>);
    const error = new Error('test error');
    wrapper.find(MockChild).simulateError(error);
    expect(wrapper.find("h1").text()).toBe('Something went wrong.');
  });
});


