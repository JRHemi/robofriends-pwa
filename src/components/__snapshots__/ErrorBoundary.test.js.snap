// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary expects to render Child component 1`] = `<MockChild />`;
